src/main/java/frc/robot/Robot.java

```diff
@@ -398,6 +400,28 @@ private void handleState() {
       m_isInControlPanelMode =
           m_entryControlPanelMode.getBoolean(m_isInControlPanelMode);
     }
+
+    // Set the Shuffleboard control panel values to their defaults when not enabled.
+    if (!m_isInControlPanelMode
+        && m_isInControlPanelModeLastLoop != m_isInControlPanelMode) {
+      m_detectedColorString = "N/A";
+      m_lastDetectedColorString = "N/A";
+      m_targetControlPanelColor = "N/A";
+      m_controlPanelSpinAmount = 0;
+      m_entryDetectedColor.setString(m_detectedColorString);
+      m_entryTargetColor.setString(m_targetControlPanelColor);
+      m_entryTargetSpin.setDouble(m_controlPanelSpinAmount);
+      m_entryConfidence.setDouble(0);
+    }
+
+    // If control panel mode is enabled and the robot is driven, disable it.
+    if ((m_controllerDrive.getRawAxis(DriveStation.kIDAxisLeftY) > 0.5
+        || m_controllerDrive.getRawAxis(DriveStation.kIDAxisRightX) > 0.5)
+        && m_isInControlPanelMode) {
+      m_isInControlPanelMode = false;
+      m_entryControlPanelMode.setBoolean(m_isInControlPanelMode);
+    }
```

Comment:
This should be moved before the last `if` block. That `if` should be ran after the mode processing has already been done.